### PR TITLE
feat(billing): Update on-demand audit log to support new on-demand budgets

### PR DESCRIPTION
**Pull request by dashed**
The new on-demand audit log entries will be generated in https://github.com/getsentry/getsentry/pull/6926 (and later in the AM checkout endpoint).

New audit log entries with on-demand budget support, and current on-demand audit log entries that was added in https://github.com/getsentry/sentry/pull/30054

Screen Shot 2022-01-27 at 7 43 49 PM

Legacy on-demand audit log entry is still supported for backwards compatibility (prior to https://github.com/getsentry/sentry/pull/30054):

Screen Shot 2022-01-27 at 11 01 12 PM

### CODE REVIEW

File: src/sentry/models/auditlogentry.py [hunk 1]
```diff
@@ -13,6 +13,24 @@
 MAX_ACTOR_LABEL_LENGTH = 64
 
 
+def format_ondemand_budget(ondemand_data):
+    if type(ondemand_data) is dict:
+        budget_mode = ondemand_data.get("budgetMode", None)
+        if budget_mode == "shared":
+            shared_budget = ondemand_data.get("sharedMaxBudget", 0)
+            return f"shared on-demand of {format_ondemand_max_spend(shared_budget)}"
+        else:
```
Comment (markstory) [inline src/sentry/models/auditlogentry.py:22]: Should we use shared as the fallback, or explicitly check for per-category? There are no formal promises/contracts on audit log data and we could end up with different data overtime.

File: src/sentry/models/auditlogentry.py [hunk 2]
```diff
@@ -369,11 +387,14 @@ def get_note(self):
...
                 return "changed on-demand spend to unlimited"
-            next_ondemand_max_spend = format_ondemand_max_spend(self.data["ondemand"])
+            next_ondemand_budget = format_ondemand_budget(self.data["ondemand"])
             if "prev_ondemand" in self.data:
-                prev_ondemand_max_spend = format_ondemand_max_spend(self.data["prev_ondemand"])
-                return f"changed on-demand max spend from {prev_ondemand_max_spend} to {next_ondemand_max_spend}"
-            return f"changed on-demand max spend to {next_ondemand_max_spend}"
+                ondemand_label = "on-demand max spend"
+                if type(self.data["ondemand"]) is dict or type(self.data["prev_ondemand"]) is dict:
+                    ondemand_label = "on-demand budget"
```
Comment (markstory) [inline src/sentry/models/auditlogentry.py:394]: Couldn't we always use the term `on-demand budget`? It is a suitable term for both on-demand implementations.

File: src/sentry/models/auditlogentry.py (hunk 1 above)
Comment (dashed) [reply src/sentry/models/auditlogentry.py]: I've swapped the if-else blocks

**Comment by dashed** [comment]
I've updated the screenshots in the pull request description with the updated changes.

**Review by markstory — APPROVED** [review]

**Pull request merged**